### ISSUE TITLE
toggle-window hotkey doesn't work under wayland

### ISSUE THREAD

**Issue by louisrubet** [Eugeny/tabby#5207]
**Describe the problem**:
Pressing a hotkey to toggle tabby doesn't work under linux / wayland when tabby is not focused, while it is working well under x11.
[gnome 41 / wayland on fedora 35, tabby 1.0.168].

Unfortunetely this is a common issue about global hotkeys under wayland.

From Jimmy Kloss: "To control a not focussed app (aka global hotkey) [your app] would require to [...] expose a dbus interface for remote control [...]". This is for example the method chosen by guake to get its hotkeys.

Strangely the hotkey is well dispatched to tabby when Chrome (Version 96) is focused. Close or unfocus Chrome and it doesn't work anymore.

**To Reproduce**:
[Steps to reproduce the behavior]

- close or hide all the visible windows on your desktop
- run tabby
- configure a hotkey for 'toggle-window'
- first press makes tabby invisible (while still running)
- second press does nothing

**Comment by harpreetsb** [comment]
any solution?

**Comment by czadikem** [comment]
Is there a fix for this yet?

**Comment by l10r** [comment]
Yes, if you're using gnome just add custom shortcut over the settings, and make it run "tabby toggle"
![image](https://user-images.githubusercontent.com/45217204/186176708-a613975b-66d4-40ba-af29-ee885dcf6947.png)
![image](https://user-images.githubusercontent.com/45217204/186176862-41adcb1d-3805-4d80-8160-ff177afc94c4.png)

It only works for showing, not for hiding sadly...

**Comment by l10r** [comment]
If the dev could make this approach actually do the other way around, it could be a good workaround.

**Comment by bernardolm** [comment]
Don't works for me :(

**Comment by bernardolm** [comment]
> Yes, if you're using gnome just add custom shortcut over the settings, and make it run "tabby toggle" ![image](https://user-images.githubusercontent.com/45217204/186176708-a613975b-66d4-40ba-af29-ee885dcf6947.png) ![image](https://user-images.githubusercontent.com/45217204/186176862-41adcb1d-3805-4d80-8160-ff177afc94c4.png)
> 
> It only works for showing, not for hiding sadly...

Don't works for me :(

**Comment by midnight-wonderer** [comment]
Hey
Adding `tabby toggle` command seems to be a good idea to me; I mean actually implementing the feature using dBus interface. (as Louis mentioned)
The current approach by l10r is equivalent to running `tabby` without argument.

How about we settle some details first, in the case when someone wants to take a shot?  
Waiting for maintainers to take on this is a distant dream; apparently, they don't run tabby on Wayland.

Question to maintainers:  
* Does `tabby toggle` sound good to you?  
FYI, dBus is kinda like a message system for Linux desktops. The idea is to handle the `toggle` argument by sending a command via `dBus` to the main window of tabby. The user on Wayland can config global hotkeys as mentioned by l10r.
* Where (and how) should we put the command line arguments handler?  
The code should `import { app } from "electron";` and read the arguments.
* Similar to above, where should we implement `dBus` handler (listener)?

**Comment by Apxdono** [comment]
One way I finally made it work on Arch KDE Wayland while there's no proper DBus support (should in theory work with other Window Managers and DEs):

1. Install `xvkbd` (for Arch it can be found in `AUR` under `xvkbd`)
1. Set `Ctrl+Space` as the "Toggle hotkey" inside Tabby
1. Created a script `toggle_tabby.sh` with this content:
    ```sh
    #/bin/sh
    xvkbd -window tabby -text "\C " # "\C " is Ctrl+Space, change to your key combination
    ```
1. Run `chmod +x toggle_tabby.sh` 
1. KDE System Settings -> Shortcuts -> `+ Add command` and choose `toggle_tabby.sh`.
1. Select new command and assign `Ctrl+Space` as the shortcut
1. Profit

**Comment by louisrubet** [comment]
Another workaround for Gnome users is the `quake-mode` Gnome extension, which permits to toggle any window and configure things like size, focus, placement etc. Convenient and well integrated, finely works on Wayland. This is the solution I use.

**Comment by func0der** [comment]
The workaround with the global shortcut does not work for AppImage users though.

**Comment by BeegMan27** [comment]
any fix for KDE? or does the gnome fix work thitherfor as well
